### PR TITLE
Streaming uploads

### PR DESCRIPTION
**Pull request by xloem**
This is an extension on #36 to stream all files.

python-requests streams file objects and iterators if they are passed as post body content.  I've changed all single file uploads to upload in that way.

### CODE REVIEW

**Comment by m-cat** [comment]
Merge conflict needs to be resolved

**Comment by xloem** [comment]
Thanks.  Resolved.

File: tests/test_integration_upload.py [hunk 1]
```diff
@@ -13,6 +13,23 @@
 client = skynet.SkynetClient()
 
 
+def response_callback(request):
```
Comment (m-cat) [inline tests/test_integration_upload.py:16]: I don't understand what this function does or why we started using it as a callback. Can you please add a docstring for the function and some comments?

File: tests/test_integration_upload.py [hunk 2]
```diff
@@ -36,16 +52,17 @@ def test_upload_file():
     print("File upload successful, sialink: " + sialink2)
 
     headers = responses.calls[0].request.headers
-    assert headers["Content-Type"].startswith("multipart/form-data;")
+    assert headers["Content-Type"]
```
Comment (m-cat) [inline tests/test_integration_upload.py:55]: q: Why did this change?

File: tests/test_integration_upload.py [hunk 3]
```diff
@@ -36,16 +52,17 @@ def test_upload_file():
...
     assert headers["User-Agent"] == "python-requests/2.24.0"
     assert "Authorization" not in headers
 
+    params = responses.calls[0].request.params
+    assert params["filename"] == "file1"
+
     body = responses.calls[0].request.body
-    assert str(body).find('Content-Disposition: form-data; name="file"; \
```
Comment (m-cat) [inline tests/test_integration_upload.py:44]: q: Why was this removed?

**Review by m-cat — CHANGES_REQUESTED** [review]
One requested change plus some questions.

**Comment by xloem** [comment]
Sorry about all that, looks like I made an error rebasing.  I think only the first change is relevant and don't immediately recognise the others.  Consider a draft until addressed.

But if you want streaming to work basically you just pass a file object straight to a single file post just like with chunked uploading.

File: tests/test_integration_upload.py (hunk 1 above)
Comment (xloem) [reply tests/test_integration_upload.py:16]: it's a way responses has for handling requests.  it could be used to do server actions like providing the appropriate skylink and response for the data.  i use it to simply gather the uploaded data so it can be verified the code under test uploaded the right data.

File: tests/test_integration_upload.py (hunk 2 above)
Comment (xloem) [reply tests/test_integration_upload.py:67]: requests doesn't do multipart streaming upload, so single file uploads are no longer multipart

File: tests/test_integration_upload.py (hunk 3 above)
Comment (xloem) [reply tests/test_integration_upload.py:44]: same reason as above: the content data no longer has multipart decorations

**Comment by xloem** [comment]
Sorry for misleading comment.  The changes are to make the tests pass, after I changed single file uploads to be streaming.  I also added tests that the body content is correct.

I added the comments you requested.

**Review by m-cat — APPROVED** [review]
Looks great. Thanks for the comments 🙂
